### ISSUE TITLE
Nodes Agent is Offline

### ISSUE THREAD

**Issue by victor-hotel** [codership/galera-manager-support#111]
Galera Manager Version: 1.8.4
Debian 12

**Nodes** 
Debian 12
MariaDB 10.11

Nodes Agent is Offline:

**Comment by victor-hotel** [comment]
![image](https://github.com/user-attachments/assets/cb3b053a-7eab-4e39-92c8-82b71ea78f1a)

**Comment by esumin** [comment]
Hello,

Could you please provide more details about your setup?

1. Did the setup work for you previously, or has the agent been offline continuously?

2. What does your setup look like? Specifically, which infrastructure provider are you using (e.g., DigitalOcean, AWS EC2)? Does your GMD host have a public IP address?

3. Are you able to SSH into any of your hosts?

    3.a. Could you please share the content of the `/etc/default/gma` file? (Please mask any sensitive information.)
    
    3.b. Could you check the status of the agent using the command `service gma status` and share the output?
    
    3.c. Could you also send the agent logs by running `cat /var/log/syslog | grep gma`? Again, please mask any sensitive information.

Thank you!

**Comment by sector7gp** [comment]
Hi! Same Here, Debian 12 and Maria 10, fresh install

![Captura de pantalla 2024-08-23 085421](https://github.com/user-attachments/assets/f528f385-640a-4634-9698-05c8907e0f87)

1) Allways have the RED DOT
2) OnPremises cluster with 5 Nodes, the GM is in an independent VM
3) Yes all of them, also tried with no firewall
3.a)
```
#cat /etc/default/gma
ARGS=""
GMA_GMD_GRPC_SCHEME="https"
GMA_GMD_GRPC_ADDRESS="clustermanager.mydomain.com:443"
GMA_AGENT_SECRET_ID="someSecret"
GMA_CLUSTER_ID="1"
GMA_NODE_ID="9"
```
3b. 
```
gma.service - gma - galera manager agent
     Loaded: loaded (/lib/systemd/system/gma.service; enabled; preset: enabled)
     Active: active (running) since Wed 2024-08-21 08:25:58 -03; 2 days ago
   Main PID: 691 (gma)
      Tasks: 12 (limit: 19133)
     Memory: 24.9M
        CPU: 17.499s
     CGroup: /system.slice/gma.service
             └─691 /usr/local/bin/gma run
```

3c. On Debian12 theres no syslog by default.

**Comment by esscz** [comment]
Hello,

AFAIK, on Debian12 you can use `journalctl -u gma.service` to see the logs.

**Comment by victor-hotel** [comment]
```
root@###################:~# journalctl -u gma.service
-- No entries --
root@###################:~# service gmd status
● gmd.service - gmd - galera manager daemon
     Loaded: loaded (/lib/systemd/system/gmd.service; enabled; preset: enabled)
     Active: active (running) since Mon 2024-08-19 15:49:59 BST; 1 week 0 days ago
   Main PID: 513 (gmd)
      Tasks: 11 (limit: 4643)
     Memory: 115.6M
        CPU: 1min 38.126s
     CGroup: /system.slice/gmd.service
             └─513 /usr/bin/gmd run
```


```
(*Nodes).checkAgentStatus","host-id":"1","level":"info","msg":"Checking agent connection","node-id":"1","time":"2024-08-19T12:54:07+01:00"}
{"_context":{"jobID":"233"},"channel-type":"app","cluster-id":"2","file":"/go/pkg/internal/mgmt/nodes.go:700","func":"github.com/codership/galera-manager/pkg/internal/mgmt.(*Nodes).checkAgentStatus","host-id":"1","level":"error","msg":"No agent connection","node-id":"1","time":"2024-08-19T12:54:07+01:00"}
{"_context":{"jobID":"233"},"channel-type":"app","cluster-id":"2","file":"/go/pkg/internal/mgmt/nodes.go:624","func":"github.com/codership/galera-manager/pkg/internal/mgmt.(*Nodes).Status.func1","host-id":"1","level":"info","msg":"Agent is not responding, attempting to (re)install","node-id":"1","time":"2024-08-19T12:54:07+01:00"}
{"_context":{"jobID":"234"},"channel-type":"app","cluster-id":"2","file":"/go/pkg/internal/mgmt/nodes.go:525","func":"github.com/codership/galera-manager/pkg/internal/mgmt.(*Nodes).InstallOrUpdateGMA.func1","host-id":"1","level":"info","msg":"installing gma","node-id":"1","time":"2024-08-19T12:54:07+01:00"}
```

Is this Telegraf metrics collection as i have metrics on the dashboard?

**Comment by victor-hotel** [comment]
> Hello,
> 
> Could you please provide more details about your setup?
> 
> 1. Did the setup work for you previously, or has the agent been offline continuously?
> 2. What does your setup look like? Specifically, which infrastructure provider are you using (e.g., DigitalOcean, AWS EC2)? Does your GMD host have a public IP address?
> 3. Are you able to SSH into any of your hosts?
>    3.a. Could you please share the content of the `/etc/default/gma` file? (Please mask any sensitive information.)
>    3.b. Could you check the status of the agent using the command `service gma status` and share the output?
>    3.c. Could you also send the agent logs by running `cat /var/log/syslog | grep gma`? Again, please mask any sensitive information.
> 
> Thank you!

I only have a GMD file:

```
ARGS=
GMD_CONFIG_DIR=/var/lib/gmd
GMD_LOGS_DIR=/var/log/gmd
INFLUXDB_URL=http://gmd:xxxxxxxxxxxxx@xxx.xxx.xxx.xxx:8081/gmd
GMD_URL=http://xxx.xxx.xxx.xxx/
```

**Comment by esscz** [comment]
You are currently checking on the GMD host, but you should instead check on specific node(s). 
To SSH into them, click on the desired node, and then go to the Console tab. 
There, you'll find a terminal window for the node.

**Comment by victor-hotel** [comment]
Ahh understood,

Hopefully the issue clearer now:

```
Aug 28 13:28:54 db01 gma[7003]: time="2024-08-28T13:28:54+01:00" level=info msg="Creating agentcom stream..."
Aug 28 13:28:54 db01 gma[7003]: time="2024-08-28T13:28:54+01:00" level=error msg="Failed to create stream: rpc error: code = Unavailable desc = connection error: desc = \"error reading server preface: http2: frame too large\""
Aug 28 13:28:54 db01 gma[7003]: time="2024-08-28T13:28:54+01:00" level=error msg="Error while serving: rpc error: code = Unavailable desc = connection error: desc = \"error reading server preface: http2: frame too large\"; sleeping for 1 second"
```

**Comment by victor-hotel** [comment]
Perhaps linked to https://github.com/codership/galera-manager-support/issues/90

**Comment by esumin** [comment]
Hi @victor-hotel, thank you for the logs. Yes, you’re correct; this is a known issue. We are actively working on a fix and will update issue #90 as soon as the fix is available.

**Comment by esscz** [comment]
Hi @victor-hotel , thank you for the logs. Yes, you’re correct; this is a known issue. We are actively working on a fix and will update issue #90 as soon as the fix is available.

**Comment by byte** [comment]
Closing this on @esumin request, as it is truly linked to #90 which will be fixed asap

Thank you all for the leg work!
